GitUI/Editor/FileViewerInternal.cs

```diff
@@ -387,6 +388,28 @@ public void AddTextHighlighting()
             _textHighlightService.AddTextHighlighting(TextEditor.Document);
         }
 
+        /// <summary>
+        /// Get the next/previous line for the grep match.
+        /// </summary>
+        /// <param name="next"><c>true</c> if next position, <c>false</c> if previous.</param>
+        /// <returns><c>true</c> if.</returns>
+        public bool GetMatchLine(bool next)
+        {
+            if (_textHighlightService is not GrepHighlightService grepHighlightService)
+            {
+                return false;
+            }
+
+            if (!(grepHighlightService.GetGrepLineNum(LineAtCaret, next) is int line) || line < 0)
+            {
+                return true;
```

Comment:
This will answer whether the second `return` shall really be `true`.

Reply:
Search is simplified in #11628, special case for grep is removed

Reply:
I could remove search in this PR to reduce noise if you prefer.

Reply:
I should have started with "xmldoc is incomplete". This was the main point.

Reply:
Navigate is still included